src/canopy/knowledge_base/knowledge_base.py

```diff
@@ -26,45 +22,46 @@
 from canopy.models.data_models import Query, Document
 
 INDEX_NAME_PREFIX = "canopy--"
-TIMEOUT_INDEX_CREATE = 300
-TIMEOUT_INDEX_PROVISION = 30
+TIMEOUT_INDEX_CREATE = 30
+TIMEOUT_INDEX_PROVISION = 300
 INDEX_PROVISION_TIME_INTERVAL = 3
 RESERVED_METADATA_KEYS = {"document_id", "text", "source"}
 
 DELETE_STARTER_BATCH_SIZE = 30
-
 DELETE_STARTER_CHUNKS_PER_DOC = 32
 
 
-def connect_to_pinecone():
-    """
-    Connect to Pinecone.
-    This method is called automatically when creating a new KnowledgeBase object.
-    Or when calling `list_canopy_indexes()`.
-    """
-    try:
-        pinecone_init()
-        pinecone_whoami()
-    except Exception as e:
-        raise RuntimeError("Failed to connect to Pinecone. "
-                           "Please check your credentials and try again") from e
+@lru_cache(maxsize=1)
+def _get_global_client() -> Pinecone:
+    return Pinecone()
 
 
-def list_canopy_indexes() -> List[str]:
+def list_canopy_indexes(pinecone_client: Pinecone = None) -> List[str]:
     """
     List all Canopy indexes in the current Pinecone account.
 
     Example:
         >>> from canopy.knowledge_base import list_canopy_indexes
         >>> list_canopy_indexes()
             ['canopy--my_index', 'canopy--my_index2']
+    Args:
+        pinecone_client: Pinecone client used to query indexes. If not passed,
+                         the global client will be used.
 
     Returns:
         A list of Canopy index names.
     """
+    pinecone_client = pinecone_client or _get_global_client()
+
+    try:
+        indexes = pinecone_client.list_indexes()
+    except Exception as e:
+        raise RuntimeError("Failed to connect to Pinecone. "
+                           "Please check your credentials and try again") from e
+
+    index_names = [_['name'] for _ in indexes.index_list['indexes']]
```

Comment:
```suggestion
    index_names = indexes.names()
```
Apparently there's a nice method for this

Reply:
Nice they added recently I guess will change to it